compiler/rustc_privacy/src/errors.rs

```diff
@@ -87,5 +90,13 @@ pub struct FromPrivateDependencyInPublicInterface<'a> {
 pub struct PrivateInPublicLint<'a> {
     pub vis_descr: &'static str,
     pub kind: &'a str,
-    pub descr: String,
+    pub descr: FromDisplay<'a>,
+}
+
+pub struct FromDisplay<'a>(pub &'a dyn Display);
```

Comment:
We could put this in `rustc_errors` or wherever the other `IntoDiagnosticArg` impls are, it could be used more widely than just in `rustc_privacy`.